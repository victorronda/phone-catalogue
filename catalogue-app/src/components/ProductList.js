import React from 'react';
import Product from './Product';

const ProductList = ({phones}) => {

    return (
        <div>
            {phones.map((phone,index) => (<Product phone={phone} key={index} />))}
        </div>
    );
}

export default ProductList;
