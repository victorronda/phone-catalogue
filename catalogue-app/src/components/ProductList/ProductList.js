import React from 'react';
import Product from '../Product/Product';


const ProductList = ({phones}) => {

    return (
        <div className="d-flex flex-row flex-wrap justify-content-center w-100 mx-auto">
            {phones.map((phone,index) => (<Product phone={phone} key={index} />))}
        </div>
    );
}

export default ProductList;
