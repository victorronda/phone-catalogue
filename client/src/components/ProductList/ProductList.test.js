import { shallow } from 'enzyme';
import React from 'react';
import ProductList from './ProductList';

it('should render ProductList component', () => {
    const mockPhones = [
        {
            id: 5,
            name: 'Nokia',
            ram: 2,
            imageFileName: 'http://localhost:4000/images/IPhone_7.png'
        },
        {
            id: 2,
            name: 'Apple',
            ram: 4,
            imageFileName: 'http://localhost:4000/images/Galaxy_S7.png'
        }

    ];
    expect(shallow(<ProductList phones={mockPhones} />)).toMatchSnapshot();
});