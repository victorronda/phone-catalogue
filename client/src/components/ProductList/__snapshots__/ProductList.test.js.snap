// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProductList component 1`] = `ShallowWrapper {}`;
